api/server/services/auth.service.js

```diff
@@ -62,61 +57,58 @@ const registerUser = async (user) => {
     const existingUser = await User.findOne({ email });
 
     if (existingUser) {
-      log({
-        title: 'Register User - Email in use',
-        parameters: [
-          { name: 'Request params:', value: user },
-          { name: 'Existing user:', value: existingUser }
-        ]
-      });
-      response = { status: 422, message: 'Email is in use' };
-      return response;
+      console.info(
+        'Register User - Email in use',
+        { name: 'Request params:', value: user },
+        { name: 'Existing user:', value: existingUser }
+      );
+      
+      // Sleep for 1 second
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+
+      // TODO: We should change the process to always email and be generic is signup works or fails (user enum)
+      return { status: 500, message: 'Something went wrong' };
```

Comment:
I have to disagree with the idea of making this generic because its a poor user experience to just tell them that "something went wrong", and will inevitably lead to them pinging us in the discord. If they know that the reg failed because the email is already in use, they can take take the appropriate action.

Reply:
@danorlando user enumeration is the reason. Generic errors prevent enum. Some people don't care about this security facet but I worry the people deploying won't know whats wrong. 
An alternative idea is sending an email, "Hey did you forget your password?" sort of thing, then you avoid enum

Reply:
I understand your point, but if someone knows how to deploy the app where it is exposed to the internet, the onus is on them to use strong passwords and implement rate limiting and whatever other security measures are necessary for production deployment. I don't think we should sacrifice user experience on our end because of the potential for some dummy to use a crappy password on a deployed instance of an open source app. I think it would be better if we instead strengthened the password requirements to perhaps 10 characters, upper and lower, special character, and number.

Reply:
I'm not sure its that much of a sacrifice?

Flow would be:
Get an error about your login, try to reset pw, you get an email either saying the account doesn't exist or it does and how to reset.

Additional flow, if you attempt to login but use the wrong pw, an email gets sent saying "are you trying to login? click here to reset". or if they don't exist, "click here to register now".

When it comes to pw restrictions, they don't actually suggest that sort of complexity anymore as it doesn't make things more secure. all lower case with 4 words is recommended over 10 characters mixed. Even apples generated pw's don't have uppercase much anymore. There are definitely ways to avoid enum without loosing the notification side of things. Maybe a discussion for the discord at any rate. I'll leave till last if its contentious but honestly I don't see a problem with it, especially with the flows mentioned

Reply:
Also, most people have pw managers these days anyway

Reply:
ok, my only question would be along the lines of - how does the user in fact know that they forgot their password or are using the wrong password and that the error is not due to some other problem? 500 means Internal server error, and is almost always the result of an unhandled exception. If I ever get this response at work, it basically means I have to go create a ticket for the backend dev who wrote the service to properly handle the exception and send me something useful back so I can properly handle it on the frontend. That's essentially the context from where this discussion is coming from. 

If we should not be telling the user that they are using invalid credentials from the UI, I'd be ok with sending a 400 (Bad Request) back and modifying the UI to no longer display the message that gets sent back from the server and instead add a line saying that there was a problem with their login attempt and to check their email for more information. It would be good to also have a ? icon as well, that when hovered, shows a tooltip stating that "For security, we do not display detailed error information here. Please check your email to find out more about why you are seeing this."

Reply:
I guess another option would be to just add rate limiting on our end, such that after 3 attempts we lock them out and force a password reset from their email, although thats a little bit more work.

Reply:
Nothing wrong with having it different from bad login vs 500. Didn't realised I changed the code